device/mellanox/x86_64-mlnx_msn2700-r0/psu_sensors_conf_updater

```diff
@@ -0,0 +1,85 @@
+    local SENSORS_CONFIG_FILE="/tmp/sensors.conf"
+    local PSU_SENSORS_CONFIG="/usr/share/sonic/platform/psu_sensors.conf"
+
+    if [ -f $SENSORS_CONFIG_FILE ]; then
+        rm $SENSORS_CONFIG_FILE
+    fi
+
+    if [ -z "$1" ]; then
+        echo "ERROR: no sensors configuration file had been selected"
+        return
+    else
+        cp $1 $SENSORS_CONFIG_FILE
+    fi
+
+    platform=$(cat /usr/share/sonic/platform/platform.json | grep -m 1 "name" | cut -d '"' -f 4)
+    # built-in psu, no need for dynamic configruation
+    if [[ "$platform" == "MSN2100" || "$platform" == "MSN2010" ]]; then return; fi
+    # incase we don't have this platform's info
+    if [[ $(jq -r --arg platform "$platform" '.platform[$platform]' PSU_SENSORS_CONFIG) == "null" ]]; then return; fi
+
+    local rev=""
+    case $(echo $1 | rev | cut -d/ -f 1| rev) in 
+        "sensors.conf")             rev="default";;
+        "sensors.conf.a1")          rev="a1";;
+        "sensors_respin.conf")      rev="respin";;
+        "sensors_swb_respin.conf")  rev="swb_respin";;
+        *)                          echo "no such hardware revision ";;
+    esac
+    if [ ! $rev ]; then return; fi
+
+    start_marker_number=$(grep -n "Power supplies" $SENSORS_CONFIG_FILE |  cut -f1 -d:)
+    end_marker_number=$(($(tail -n +$(($start_marker_number + 1)) $SENSORS_CONFIG_FILE | grep -n '^$' | head -n 1 | cut -d: -f1) + $start_marker_number))
+    # In some cases, the PSU sensor section may be at the end of the file,
+    # so using grep alone may not match it. Let's set the end marker
+    # to the total number of lines in the file directly.
+    if [ "$start_marker_number" == "start_marker_number" ]; then
+        end_marker_number=$(wc -l $SENSORS_CONFIG_FILE)
+    fi
+    
+    sed -i "${start_marker_number},${end_marker_number}d" $SENSORS_CONFIG_FILE
+
+    echo "" >> $SENSORS_CONFIG_FILE
+    echo "# Power supplies" >> $SENSORS_CONFIG_FILE
+
+    bus=$(jq -r --arg platform "$platform" --arg rev "$rev" '.platform[$platform][$rev].bus | map("\"" + . + "\"") | join(" ")' PSU_SENSORS_CONFIG)
+    if [ "$bus" ]; then echo "bus $bus" >> $SENSORS_CONFIG_FILE; fi
+
+    mapfile -t chips < <(jq -r --arg platform "$platform" --arg rev "$rev" '.platform[$platform][$rev].chip | to_entries[] | .key' PSU_SENSORS_CONFIG )
+
+    for chip in "${chips[@]}"; do
+        echo "    chip "$chip"" >> $SENSORS_CONFIG_FILE
+        number=$(jq -r --arg platform "$platform" --arg rev "$rev" --arg chip "$chip" '.platform[$platform][$rev].chip[$chip] | to_entries | .[0].value' PSU_SENSORS_CONFIG)
+        side=$(jq -r --arg platform "$platform" --arg rev "$rev" --arg chip "$chip" '.platform[$platform][$rev].chip[$chip] | to_entries | .[1].value' PSU_SENSORS_CONFIG)
+        psu=$(cat /var/run/hw-management/eeprom/psu${number}_vpd | grep "PN_VPD_FIELD" | cut -d ' ' -f 2 2>&1)
+        if [ ! "$psu" ]; then
+            echo "ERROR: Failed to read from /var/run/hw-management/eeprom/psu${number}_vpd"
```

Comment:
Shall we add some more info about the script's name to indicate that the log is from this script?

Reply:
added the script's name as a log identifier